skbio/io/format/ordination.py

```diff
@@ -232,32 +233,32 @@ def _ordination_to_ordination_results(fh):
                                   'proportion explained values')
     _check_empty_line(fh)
 
-    species, species_ids = _parse_array_section(fh, 'Species')
+    species = _parse_array_section(fh, 'Species')
     _check_length_against_eigvals(species, eigvals,
                                   'coordinates per species')
     _check_empty_line(fh)
 
-    site, site_ids = _parse_array_section(fh, 'Site')
+    site = _parse_array_section(fh, 'Site')
     _check_length_against_eigvals(site, eigvals,
                                   'coordinates per site')
     _check_empty_line(fh)
 
     # biplot does not have ids to parse (the other arrays do)
-    biplot, _ = _parse_array_section(fh, 'Biplot', has_ids=False)
+    biplot = _parse_array_section(fh, 'Biplot', has_ids=False)
     _check_empty_line(fh)
 
-    cons, cons_ids = _parse_array_section(fh, 'Site constraints')
+    cons = _parse_array_section(fh, 'Site constraints')
 
-    if cons_ids is not None and site_ids is not None:
-        if cons_ids != site_ids:
+    if cons is not None and site is not None:
+        if not np.array_equal(cons.index, site.index):
             raise OrdinationFormatError(
                 "Site constraints ids and site ids must be equal: %s != %s" %
-                (cons_ids, site_ids))
+                (cons.index, site.index))
 
     return OrdinationResults(
-        eigvals=eigvals, species=species, site=site, biplot=biplot,
-        site_constraints=cons, proportion_explained=prop_expl,
-        species_ids=species_ids, site_ids=site_ids)
+        short_method_name='', long_method_name='', eigvals=eigvals,
```

Comment:
For the time being, I'd vote no - just because the file format is still a working progress.  
But I do think that this should be added in once we start hammering down the specifics of the file format - so we'll probably want to write up an issue for this.